sdk/core/azure-core/inc/azure/core/internal/diagnostics/log.hpp

```diff
@@ -157,10 +112,16 @@ namespace Azure { namespace Core { namespace Diagnostics { namespace _internal {
        * @tparam T Type of the object being inserted.
        * @param val value to be inserted into the underlying stream.
        */
-      template <typename T> std::ostream& operator<<(T val) { return m_stream << val; }
+      template <typename T> std::ostream& operator<<(T val)
+      {
+        std::unique_lock<std::mutex> lock(m_mutex);
+        return m_stream << val;
+      }
```

Comment:
```suggestion
      template <typename T> Stream& operator<<(T val)
      {
        std::unique_lock<std::mutex> lock(m_mutex);
        m_stream << val;
        return *this;
      }
```

if you don't do this, then the second call to `operator<<` gets a reference to `m_stream` and writes to it without any locks.

BTW, maybe it is reasonable to not have a lock here for now at all?
Based on how our code uses the log stream, all access to it is via unnamed local variable and within the same thread.

Reply:
Interesting point. FWIW, this doesn't compile because *this isn't a std::ostream - it wraps a std::ostream instead.

Which in turn makes it extremely clear that the lock provided no value to the codebase. Removing it.